internal/storage/inmem/watch.go

```diff
@@ -200,20 +209,30 @@ func (s *Store) watchSnapshot(req stream.SubscribeRequest, snap stream.SnapshotA
 		return 0, nil
 	}
 
-	events := make([]stream.Event, len(results))
-	for i, r := range results {
-		events[i] = stream.Event{
+	events := make([]stream.Event, 0, len(results)+2)
+	addEvent := func(event *pbresource.WatchEvent) {
+		events = append(events, stream.Event{
 			Topic: eventTopic,
 			Index: idx,
 			Payload: eventPayload{
 				subject: req.Subject,
-				event: &pbresource.WatchEvent{
-					Operation: pbresource.WatchEvent_OPERATION_UPSERT,
-					Resource:  r,
-				},
+				event:   event,
 			},
-		}
+		})
+	}
+
+	addEvent(&pbresource.WatchEvent{
```

Comment:
I wish there was a way to use `Watch.nextEvent` 's witnessing of the `e.IsFramingEvent()` stream info to propagate back to the caller, but since it takes a trip through a different grpc setup that only allows `Operation` values through sadly I think this bonus framing under the framing is probably our best bet.